### ISSUE TITLE
BUG-000107223  Error while running the Viewshed tool from the Military Tools for ArcGIS in ArcMap 10.5.1

### ISSUE THREAD

**Issue by lfunkhouser** [Esri/visibility-addin-dotnet#222]
<div><div>
<span><b>Salesforce ID:</b> BUG-000107223</span><br/>
<span><b>Salesforce Submitter:</b> Sayak Nag</span><br/>
<span><b>Salesforce Submit Date:</b> 8/09/2017 8:19 AM</span><br/>
<span><b>Salesforce Bug Type:</b> Failure/Error</span><br/>
<span><b>Salesforce Severity:</b> Medium</span><br/>
<span><b>Repro Data:</b> ...\sf_filestore\PRD\Attachments\Defects\BUG-000107223</span><br/>
<span><b>Work Around:</b> (n/a)</span><br/>
<span><b>Product:</b> ArcGIS for Desktop</span><br/>
<span><b>Functional Category:</b> ArcGIS for Defense</span><br/>
<span><b>Client Platform:</b> (n/a)</span><br/>
<span><b>Version Found:</b> 10.5.1</span><br/>
<span><b>Planned Version Fixed:</b> (n/a)</span><br/>
<span><b>Comment:</b> (n/a)</span><br/>
</div></div>

Steps to Repro:

1. Download Military Tools for ArcGIS from the following link: http://solutions.arcgis.com/defense/help/military-tools/
2. Followed the steps mentioned in the following documentation: http://solutions.arcgis.com/defense/help/military-tools/web-appbuilder/set-up-elevation-geoprocessing-service.htm
3. After extraction, open the Viewshed.py script located in ...\Web AppBuilder Developer Edition\GeoProcessing\Viewshed folder.
4. Replace the sample elevation path (line 19), elevation = r"d:\Workspace\data\n36prj.tif", with your source elevation data path.
5. Save and close the Viewshed.py script.
6. Open a new ArcMap 10.5.1 document (with Advanced License)
7. Enable 3D Analyst and Spatial Analyst extension.
8. Under Geoprocessing Options enable "Overwrite the outputs of geoprocessing operations".
9. Add your elevation data to the map using the Add Data button.
10. Open the Viewshed tool in the Viewshed toolbox in the ...\Web AppBuilder Developer Edition\GeoProcessing\Viewshed folder.
11. Choose Input_Observer and place a point on the map.
12. Click OK to run the tool.
13. The tool does not run successfully and throws an error.
14. Error Message:

```
Executing: Viewshed "Feature Set" 3000 40 120 20 1000 in_memory\viewshed in_memory\wedge in_memory\fullwedge
Start Time: Wed Aug 09 08:02:40 2017
Running script Viewshed...
Error: Input Observer(s) does not fall within the extent of the input surface: dem.flt!
Failed script Viewshed...

Traceback (most recent call last):
  File "C:\Users\saya0000\Desktop\MilitaryToolsforArcGIS\Web AppBuilder Developer Edition\Geoprocessing\Viewshed\Viewshed.py", line 222, in <module>
    main()
  File "C:\Users\saya0000\Desktop\MilitaryToolsforArcGIS\Web AppBuilder Developer Edition\Geoprocessing\Viewshed\Viewshed.py", line 164, in main
    raise Exception(msgErrorPointNotInSurface)
Exception: Error: Input Observer(s) does not fall within the extent of the input surface: dem.flt!

Failed to execute (Viewshed).
Failed at Wed Aug 09 08:02:40 2017 (Elapsed Time: 0.46 seconds)
```

15. Have tried the same workflow and was able to run the tool successfully on different ArcMap versions of 10.3.1, 10.4, 10.4.1 and 10.5
16. Tried running the tool with a different data.
17. Also tried changing the projection to a different Projected Coordinate System and it came up with the same error.
18. The Testing Data is attached in the Bug Attachments.

**Comment by csmoore** [comment]
Using the test data provided ( Demoraster.tif at `\sf_filestore\PRD\Attachments\Defects\BUG-000107223`) I was not able to reproduce this issue with the test data provided, however I did get another error:

![image](https://user-images.githubusercontent.com/3090809/33491709-12383780-d689-11e7-86c8-1af1e0fa76b7.png)

That is fixed in #246 - this seems to run fine for me now and should be ready to verify

**Comment by topowright** [comment]
I have verified this works
